Account/FirefoxAccount.swift

```diff
@@ -181,6 +182,19 @@ open class FirefoxAccount {
         }
     }
 
+    open func updateFxADevices(withDb db: BrowserDB) -> Success {
+        let cachedState = stateCache.value!
+        guard let session = cachedState as? TokenState else {
+            return deferMaybe(FxAClientError.local(NSError()))
+        }
+        let client = FxAClient10(endpoint: self.configuration.authEndpointURL)
+        return client.devices(withSessionToken: session.sessionToken as NSData) >>== { resp in
+            let devices = resp.devices.map { RemoteDevice(id: $0.id!, name: $0.name, type: $0.type!, isCurrentDevice: $0.isCurrentDevice, lastAccessTime: $0.lastAccessTime) }
+            let remoteDevices = SQLiteRemoteDevices(db: db)
+            return remoteDevices.replaceRemoteDevices(remoteDevices: devices)
```

Comment:
Use protocol-oriented programming here! Define a protocol for a _thing to which you can give a list of devices_, and pass that as the argument. That should get rid of your coupling to `Storage`.